src/tools/ConsoleRunner/main.cpp

```diff
@@ -97,29 +97,15 @@ int main(int argc, char** argv)
         fullsize += strlen(argv[i]);
     }
     // add some slack for spaces between args plus quotes around executable
-    fullsize += argc + 32;
+    // and terminating NUL
+    fullsize += 3 * argc + 1;
 
     char* cmdline = new char[fullsize];
-    memset(cmdline, 0, fullsize);
-
-    // 1st arg (executable) enclosed in quotes to support filenames with spaces
-    bool sp = hasSpaces(argv[1]);
-    if (sp)
-        strcat(cmdline, "\"");
-    strcat(cmdline, argv[1]);
-    if (sp)
-        strcat(cmdline, "\"");
-    strcat(cmdline, " ");
-
-    for (int i = 2; i < argc; ++i)
+    cmdline[0] = '\0';
+
+    for (int i = 1, n = 0; i < argc; ++i)
     {
-        sp = hasSpaces(argv[i]);
-        if (sp)
-            strcat(cmdline, "\"");
-        strcat(cmdline, argv[i]);
-        if (sp)
-            strcat(cmdline, "\"");
-        strcat(cmdline, " ");
+        n += sprintf(cmdline + n, hasSpaces(argv[i]) ? "\"%s\" " : "%s ", argv[i]);
```

Comment:
snprintf? Handle snprintf failure?

Reply:
Not necessary given the amended definition of `fullsize`. Before the change such a precaution would have been needed.